workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl

```diff
@@ -0,0 +1,912 @@
+cwlVersion: v1.1
+class: Workflow
+
+# Extensions
+$namespaces:
+    s: https://schema.org/
+$schemas:
+  - https://schema.org/version/latest/schemaorg-current-http.rdf
+
+# Metadata
+s:author:
+    class: s:Person
+    s:name: Sehrish Kanwal
+    s:email: sehrish.kanwal@umccr.org
+
+# ID/Docs
+id: dragen-pon-qc--3.9.3
+label: dragen-pon-qc v(3.9.3)
+doc: |
+    Documentation for dragen-pon-qc v3.9.3.
+    This workflow calls Dragen somatic tool with normal samples for creating
+    PoN. Additionally it performs QC on the output through GHIF-QC workflow.
+
+requirements:
+  MultipleInputFeatureRequirement: {}
+  InlineJavascriptRequirement: {}
+  SubworkflowFeatureRequirement: {}
+  SchemaDefRequirement:
+    types:
+      - $import: ../../../schemas/fastq-list-row/1.0.0/fastq-list-row__1.0.0.yaml
+
+# Declare inputs
+inputs:
+  # File inputs
+  tumor_fastq_list:
+    label: tumor fastq list
+    doc: |
+      CSV file that contains a list of FASTQ files
+      to process. read_1 and read_2 components in the CSV file must be presigned urls.
+    type: File?
+  tumor_fastq_list_rows:
+    label: tumor fastq list rows
+    doc: |
+      Alternative to providing a file, one can instead provide a list of 'fastq-list-row' objects for tumor sample
+    type: ../../../schemas/fastq-list-row/1.0.0/fastq-list-row__1.0.0.yaml#fastq-list-row[]?
+  reference_tar:
+    label: reference tar
+    doc: |
+      Path to ref data tarball
+    type: File
+  # Naming options
+  output_directory:
+    label: output directory
+    doc: |
+      Required - The output directory.
+    type: string
+  output_file_prefix:
+    label: output file prefix
+    doc: |
+      Required - the output file prefix
+    type: string
+  # Optional operation modes
+  # Given we're running from fastqs
+  # --enable-variant-caller option must be set to true (set in arguments), --enable-map-align is then activated by default
+  # --enable-map-align-output to keep bams
+  # --enable-duplicate-marking to mark duplicate reads at the same time
+  # --enable-sv to enable the structural variant calling step.
+  enable_map_align_output:
+    label: enable map align output
+    doc: |
+      Enables saving the output from the
+      map/align stage. Default is true when only
+      running map/align. Default is false if
+      running the variant caller.
+    type: boolean?
```

Comment:
Is it appropriate to leave this as an input to the user for this workflow? Should this be set to something by default?

Reply:
As discussed, if set to false will fail, just remove this, and set to true in step level

Reply:
fixed in https://github.com/umccr/cwl-ica/pull/85/commits/f501c4effabbf62bb55084de6998dd12789dc882